### DISCUSSION TITLE
Prevent editor from stealing focus on text node `setStyle`

### DISCUSSION THREAD

**Discussion by atomiks** [facebook/lexical#5032] [Q&A]
I have an input that I'm typing into to highlight words within the editor. Every time I use `node.setStyle(...)` the editor steals the focus from the input. How do I prevent this, or is there a better way to achieve this?

https://codesandbox.io/s/wonderful-sunset-r6zkh4?file=/src/Editor.js

**Comment by borgateo** [comment]
Friendly nudge. Any updates on this?

**Reply by acmoles** [reply]
I get similar behaviour. I'm using a number of internally developed plugins and at least one of them must be doing something similar to `node.setStyle(...)` so that focus is also stolen from an external input field.

**Comment by BenJavaCode** [comment]
Hmh this is a bit of a hack, so probably don't use it;)
But for example if you where using react you could create a ref like:

`let focusChangeBackStore = useRef<Element | undefined>(undefined);`

then in the update that "steals" the focus you can do:

```
const maybeElement = document.activeElement; // current focused element

if (maybeElement) { // for changing focus back after editor update 
  focusChangeBackStore.current = maybeElement;
}
```


and then in an effect, change back the current focus when the editor changes ( you could also use a node transform).

```
  useEffect(() => {
    editor.registerUpdateListener((editorState) => {
      if (focusChangeBackStore.current){
        (focusChangeBackStore.current as HTMLElement).focus();
        focusChangeBackStore.current = undefined;
      }
    });

  }, [editor]);

```

However, I think the focus is being changed due to a KEY_ENTER_COMMAND, so you could also register a new command that changes the focus back instead of using an updateListener.

**Comment by lukasb** [comment]
Try setting the editor non-editable before the setStyle, then making it editable again in a discrete update or in an update listener.
